WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -391,36 +386,36 @@ class PostListViewModel @Inject constructor(
         when (buttonType) {
             BUTTON_EDIT -> editPostButtonAction(site, post)
             BUTTON_RETRY -> _postListAction.postValue(RetryUpload(post))
-            BUTTON_RESTORE -> TODO()
+            BUTTON_RESTORE -> {
+                restorePost(post)
+            }
             BUTTON_SUBMIT, BUTTON_SYNC, BUTTON_PUBLISH -> {
                 showPublishConfirmationDialog(post)
             }
             BUTTON_VIEW -> _postListAction.postValue(ViewPost(site, post))
             BUTTON_PREVIEW -> _postListAction.postValue(PreviewPost(site, post))
             BUTTON_STATS -> _postListAction.postValue(ViewStats(site, post))
-            BUTTON_TRASH, BUTTON_DELETE -> {
-                showTrashConfirmationDialog(post)
+            BUTTON_TRASH -> {
+                trashPost(post)
+            }
+            BUTTON_DELETE -> {
+                showDeletePostConfirmationDialog(post)
             }
             BUTTON_MORE -> {
             } // do nothing - ui will show a popup window
             BUTTON_BACK -> TODO("will be removed during PostViewHolder refactoring")
         }
     }
 
-    private fun showTrashConfirmationDialog(post: PostModel) {
-        if (postIdToTrash != null) {
-            // We can only handle one trash action at once due to be able to undo
-            return
-        }
+    private fun showDeletePostConfirmationDialog(post: PostModel) {
         // We need network connection to delete a remote post, but not a local draft
         if (!post.isLocalDraft && !checkNetworkConnection()) {
             return
         }
         val messageRes = if (!UploadService.isPostUploadingOrQueued(post)) {
-            if (post.isLocalDraft) {
-                R.string.dialog_confirm_delete_permanently_post
-            } else R.string.dialog_confirm_delete_post
+            R.string.dialog_confirm_delete_permanently_post
         } else {
+            // TODO Can this ever happen? - delete action is disabled when an upload is in progress
```

Comment:
I have been thinking about this and I still believe it can't happen. The delete action is called only when the post is already in trash and posts in trash can't have upload in progress. I think we can just remove the `messageRes` and use `R.string.dialog_confirm_delete_permanently_post` directly.

Reply:
Fixed in 6365cc1fb77d55f77bc656024b1ef966bfffcc3a.

Reply:
This change was reverted in a2c53597808fc034deb8e475b4a8edaf11de21d8 and then the correct fix was applied back in 2e709c4f05fdf322c8b9e4d4349988e77206a0a1. Sorry for the double fix. For some reason I thought delete was only for local drafts 🤦‍♂️